File: src/main/java/com/pedigree/layout/LayoutResult.java
package com.pedigree.layout;

import java.awt.geom.Point2D;
import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Set;

/**
 * Stores layout positions for nodes by ID.
 */
public class LayoutResult {
    private final Map<String, Point2D> positions = new LinkedHashMap<>();

    public void setPosition(String nodeId, double x, double y) {
        positions.put(nodeId, new Point2D.Double(x, y));
    }

    public Point2D getPosition(String nodeId) {
        return positions.get(nodeId);
    }

    public Set<String> getNodeIds() {
        return Collections.unmodifiableSet(positions.keySet());
    }
}

File: src/main/kotlin/com/family/tree/layout/LayoutResult.kt
package com.family.tree.layout

import java.awt.geom.Point2D

/**
 * Stores layout positions for nodes by ID.
 */
public class LayoutResult {
    private val positions: MutableMap<String, Point2D> = LinkedHashMap()

    @JvmOverloads
    public fun setPosition(nodeId: String, x: Double, y: Double) {
        positions[nodeId] = Point2D.Double(x, y)
    }

    public fun getPosition(nodeId: String): Point2D? = positions[nodeId]

    public fun nodeIds(): Set<String> = java.util.Collections.unmodifiableSet(positions.keys)

    // Java interop compatibility
    @JvmName("getNodeIds")
    public fun getNodeIds(): Set<String> = nodeIds()
}
